web/src/TiltMenu.tsx

```diff
@@ -0,0 +1,297 @@
+import UpdateDialog from "./UpdateDialog"
+
+const TiltMenuRoot = styled.div`
+  flex-grow: 1;
+  display: flex;
+  align-items: stretch;
+  justify-content: flex-end;
+`
+export const MenuButton = styled.button`
+  ${mixinResetButtonStyle};
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-end;
+  align-items: center;
+  transition: color ${AnimDuration.default} ease;
+  position: relative; // Anchor MenuButtonLabel, which shouldn't affect this element's width
+  padding-top: ${SizeUnit(0.5)};
+  padding-left: ${SizeUnit(0.5)};
+  padding-right: ${SizeUnit(0.5)};
+  padding-bottom: ${SizeUnit(0.5)};
+  font-size: ${FontSize.smallest};
+  color: ${Color.blue};
+
+  & .fillStd {
+    fill: ${Color.blue};
+    transition: fill ${AnimDuration.default} ease;
+  }
+  &:hover .fillStd {
+    fill: ${Color.blueLight};
+  }
+  & .fillBg {
+    fill: ${Color.grayDarker};
+  }
+
+  &.is-disabled {
+    pointer-events: none;
+    cursor: default;
+  }
+`
+const MenuButtonLabel = styled.div`
+  position: absolute;
+  bottom: 0;
+  font-size: ${FontSize.smallest};
+  color: ${Color.blueDark};
+  width: 200%;
+  transition: opacity ${AnimDuration.default} ease;
+  opacity: 0;
+
+  ${MenuButton}:hover &,
+  ${MenuButton}[data-open="true"] & {
+    opacity: 1;
+  }
+`
+const UpdateAvailableFloatIcon = styled(UpdateAvailableIcon)`
+  display: none;
+  position: absolute;
+  top: 15px;
+  left: 5px;
+  width: 10px;
+  height: 10px;
+
+  &.is-visible {
+    display: block;
+  }
+`
+
+type TiltMenuShortcutsProps = {
+  toggleShortcutsDialog: () => void
+  snapshot: SnapshotAction
+}
+
+/**
+ * Sets up keyboard shortcuts that depend on the tilt menu.
+ */
+class TiltMenuShortcuts extends Component<TiltMenuShortcutsProps> {
+  constructor(props: TiltMenuShortcutsProps) {
+    super(props)
+    this.onKeydown = this.onKeydown.bind(this)
+  }
+
+  componentDidMount() {
+    document.body.addEventListener("keydown", this.onKeydown)
+  }
+
+  componentWillUnmount() {
+    document.body.removeEventListener("keydown", this.onKeydown)
+  }
+
+  onKeydown(e: KeyboardEvent) {
+    if (e.metaKey || e.altKey || e.ctrlKey || e.isComposing) {
+      return
+    }
+    if (e.key === "?") {
+      this.props.toggleShortcutsDialog()
+      e.preventDefault()
+    } else if (e.key === "s" && this.props.snapshot.enabled) {
+      this.props.snapshot.openModal()
+      e.preventDefault()
+    }
+  }
+
+  render() {
+    return <span></span>
+  }
+}
+
+type MetricsServing = Proto.webviewMetricsServing
+
+type TiltMenuProps = {
+  isSnapshot: boolean
+  tiltCloudUsername: string
+  tiltCloudSchemeHost: string
+  tiltCloudTeamID: string
+  tiltCloudTeamName: string
+  snapshot: SnapshotAction
+  showUpdate: boolean
+  suggestedVersion: string | null | undefined
+  runningBuild: Proto.webviewTiltBuild | undefined
+  showMetricsButton: boolean
+  metricsServing: MetricsServing | null | undefined
+}
+
+export function TiltMenu(props: TiltMenuProps) {
```

Comment:
maybe call this GlobalNav after https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav ?

By convention, I would usually interpret `TiltMenu` to be "a generic Menu component with Tilt brand colors", which is not what you want to communicate here (and even then, I think using the product name in the component name is considered bad form)